crates/provider/src/provider/trait.rs

```diff
@@ -251,8 +300,10 @@ pub trait Provider<T: Transport + Clone = BoxTransport, N: Network = Ethereum>:
     /// Gets the balance of the account.
     ///
     /// Defaults to the latest block. See also [`RpcWithBlock::block_id`].
-    fn get_balance(&self, address: Address) -> RpcWithBlock<T, Address, U256> {
-        RpcWithBlock::new(self.weak_client(), "eth_getBalance", address)
+    fn get_balance(&self, address: Address) -> ProviderCall<T, Address, U256, U256> {
+        let call = RpcWithBlock::new(self.weak_client(), "eth_getBalance", address);
+
+        ProviderCall::from(call)
```

Comment:
this does not really work, because this now erases the RpcWithBlock functions making this type a bit redundant.